allennlp/nn/beam_search.py

```diff
@@ -191,18 +181,29 @@ def new_step(
             (batch_size * self.beam_size, num_classes), float("-inf")
         )
         log_probs_after_end[:, self._end_index] = 0.0
-
         # Set the same state for each element in the beam.
         for key, state_tensor in state.items():
             if state_tensor is None:
                 continue
-            _, *last_dims = state_tensor.size()
+
             # shape: (batch_size * beam_size, *)
-            state[key] = (
-                state_tensor.unsqueeze(1)
-                .expand(batch_size, self.beam_size, *last_dims)
-                .reshape(batch_size * self.beam_size, *last_dims)
-            )
+            if state_tensor.size()[0] == batch_size:
```

Comment:
This is going to fail if the batch size is exactly equal to the number of layers, isn't it? Is there a better way to detect this, such as the number of dimensions in `state_tensor`?

Reply:
Yes, I could use the code like `state_tensor.dim() > 3` . But we use `*last_dims` for the shape. 
I wonder are there any situation that we accept tensor has dimension more than 3 ?
 If true, I have to check whether the tensors in state have the same dimension. It's a rude way :<

Reply:
fix